### ISSUE TITLE
Build FastAPI Prediction Service with ML Model

### ISSUE THREAD

**Issue by wlevan3** [wlevan3/ml-platform-engineering-practicum#2]
## Overview
Implement a simple prediction service with FastAPI that wraps a basic ML model and exposes a `/predict` endpoint. This is your **Concrete Experience** - hands-on building with minimal scaffolding.

## Context
From **Project 1: Containerized Microservice for Model Serving** (Foundations)
- **Phase**: Concrete Build & Exploration
- **Dependencies**: #1 (complete tech stack selection first)
- **Learning Objective**: Build RESTful API for ML inference, understand model serving stack

### 1. Choose and Train a Simple Model
- Use scikit-learn (e.g., RandomForestClassifier on Iris dataset)
- OR use a small neural net (PyTorch/TensorFlow)
- OR download a pre-trained model
- Save model to file (pickle, joblib, or SavedModel format)

### 2. Create FastAPI Application
```python
# Example structure:
# app/
#   ├── main.py          # FastAPI app
#   ├── model.py         # Model loading and prediction logic
#   ├── schemas.py       # Pydantic request/response models
#   └── requirements.txt
```

### 3. Implement /predict Endpoint
- Accept JSON requests with input features
- Preprocess input data
- Run model inference
- Return predictions as JSON response
- Handle errors gracefully (validation, model errors)

### 4. Add Supporting Endpoints
- `GET /health` - health check
- `GET /` - basic info about the service
- Optional: `GET /model/info` - model metadata

### 5. Local Testing
- Test with curl or Postman
- Verify predictions are correct
- Test error handling (invalid input, missing fields)

## Acceptance Criteria
- [ ] FastAPI application created with /predict endpoint
- [ ] Simple ML model trained and saved (or pre-trained model used)
- [ ] Model loads successfully on application startup
- [ ] /predict endpoint accepts JSON requests and returns predictions
- [ ] Unit tests for preprocessing and prediction logic
- [ ] Local testing successful with curl or Postman (`curl -X POST http://localhost:8000/predict -H "Content-Type: application/json" -d '{"features": [...]}'`)
- [ ] Code follows Python best practices (type hints, docstrings, Pydantic schemas)
- [ ] README with API usage examples

## Example API Contract
**Request**:
```json
POST /predict
{
  "features": [5.1, 3.5, 1.4, 0.2]
}
```

**Response**:
```json
{
  "prediction": "setosa",
  "confidence": 0.95,
  "model_version": "v1.0"
}
```

## Resources
- FastAPI Tutorial: https://fastapi.tiangolo.com/tutorial/
- Pydantic Models: https://docs.pydantic.dev/
- Scikit-learn Model Persistence: https://scikit-learn.org/stable/model_persistence.html

## Labels
- Feature Request
- Phase: Iteration 1
- Complexity: 2 (Moderate)

**Component**: 📦 Model Registry  
**Priority**: 🔴 Critical  
**Estimated Time**: 4-8 hours
